csv/shared/src/main/scala/fs2/data/csv/package.scala

```diff
@@ -342,4 +358,89 @@ package object csv {
         .stream
 
   }
+
+  object lenient {
+
+    /** Decode a char-like stream (see [[fs2.data.text.CharLikeChunks]]) into a specified type, with failures at the
+      * element level instead of failing the stream.
+      *
+      * Scenarios:
+      * - If skipHeaders is false, then the file contains no headers.
+      * - If skipHeaders is true, then the headers in the file will be skipped.
+      *
+      * For both scenarios the file is assumed to be compliant with the set of headers given.
+      */
+    def attemptDecodeGivenHeaders[T]: PartiallyAppliedDecodeAttemptGivenHeaders[T] =
+      new PartiallyAppliedDecodeAttemptGivenHeaders[T](dummy = true)
+
+    class PartiallyAppliedDecodeAttemptGivenHeaders[T](val dummy: Boolean) extends AnyVal {
+      def apply[F[_], C, Header](headers: NonEmptyList[Header],
+                                 skipHeaders: Boolean = false,
+                                 separator: Char = ',',
+                                 quoteHandling: QuoteHandling = QuoteHandling.RFCCompliant)(implicit
+          F: RaiseThrowable[F],
+          C: CharLikeChunks[F, C],
+          T: CsvRowDecoder[T, Header]): Pipe[F, C, Either[Throwable, T]] = {
```

Comment:
We can narrow these down from `Throwable` to `CsvException`, can't we? That would signal the behaviour better and allow clearer error handling.

Reply:
Yes we can! I had to change a few things in the existing API, as there it was already a `Throwable`. See new commits for the changes.